src/RunState.cc

```diff
@@ -414,20 +414,38 @@ void get_final_stats()
 		{
 		iosource::PktSrc::Stats s;
 		ps->Statistics(&s);
-		double dropped_pct = s.dropped > 0.0
-		                         ? ((double)s.dropped / ((double)s.received + (double)s.dropped)) *
-		                               100.0
-		                         : 0.0;
+
+		double dropped_pct = 0.0;
+		if ( s.dropped > 0 )
+			dropped_pct = (static_cast<double>(s.dropped) /
+			               (static_cast<double>(s.received) + static_cast<double>(s.dropped))) *
+			              100.0;
 
 		uint64_t not_processed = packet_mgr->GetUnprocessedCount();
 		double unprocessed_pct = not_processed > 0
 		                             ? ((double)not_processed / (double)s.received) * 100.0
 		                             : 0.0;
 
-		reporter->Info("%" PRIu64 " packets received on interface %s, %" PRIu64
-		               " (%.2f%%) dropped, %" PRIu64 " (%.2f%%) not processed",
-		               s.received, ps->Path().c_str(), s.dropped, dropped_pct, not_processed,
-		               unprocessed_pct);
+		if ( s.filtered )
+			{
+			double filtered_pct = 0.0;
+			if ( s.filtered && s.filtered.value() > 0 )
```

Comment:
```suggestion
			if ( s.filtered.value() > 0 )
```

Also wonder if a small lambda `pct = [](uint64_t v, uint64_t received) { ... }` would make it a bit shorter.